lte/gateway/c/core/oai/test/amf/amf_app_test_util.cpp

```diff
@@ -164,6 +164,217 @@ int send_uplink_nas_registration_complete(
+  response->imsi_length  = 15;
+  response->paa.pdn_type = IPv4;
+  inet_pton(AF_INET, "192.168.128.254", &(response->paa.ipv4_address));
+  response->paa.vlan         = 1;
+  response->pdu_session_id   = 1;
+  response->pdu_session_type = IPv4;
+  response->pti              = 0x01;
+  response->result           = 0;
+}
+
+int send_ip_address_response_itti() {
+  int rc = RETURNerror;
+
+  itti_amf_ip_allocation_response_t response = {};
+  create_ip_address_response_itti(&response);
+
+  rc = amf_smf_handle_ip_address_response(&response);
+
+  return rc;
+}
+
+void create_pdu_session_response_ipv4_itti(
+    itti_n11_create_pdu_session_response_t* response) {
+  if (!response) return;
+  std::string imsi = "222456000000001";
+  std::copy(imsi.begin(), imsi.end(), std::begin(response->imsi));
+
+  response->sm_session_fsm_state = sm_session_fsm_state_t::CREATING;
+  response->sm_session_version   = 0;
+  response->pdu_session_id       = 1;
+  response->pdu_session_type     = IPV4;
+  response->selected_ssc_mode    = SSC_MODE_3;
+  response->m5gsm_cause          = M5GSM_OPERATION_SUCCESS;
+
+  response->session_ambr.uplink_unit_type   = 0;
+  response->session_ambr.uplink_units       = 100000000;
+  response->session_ambr.downlink_unit_type = 0;
+  response->session_ambr.downlink_units     = 100000000;
+
+  response->qos_list.qos_flow_req_item.qos_flow_identifier = 9;
+  response->qos_list.qos_flow_req_item.qos_flow_level_qos_param
+      .qos_characteristic.non_dynamic_5QI_desc.fiveQI = 9;
+  response->qos_list.qos_flow_req_item.qos_flow_level_qos_param
+      .alloc_reten_priority.priority_level = 1;
+  response->qos_list.qos_flow_req_item.qos_flow_level_qos_param
+      .alloc_reten_priority.pre_emption_cap = SHALL_NOT_TRIGGER_PRE_EMPTION;
+  response->qos_list.qos_flow_req_item.qos_flow_level_qos_param
+      .alloc_reten_priority.pre_emption_vul = NOT_PREEMPTABLE;
+  response->upf_endpoint.teid[0]            = 0x7f;
+  response->upf_endpoint.teid[1]            = 0xff;
+  response->upf_endpoint.teid[2]            = 0xff;
+  response->upf_endpoint.teid[3]            = 0xff;
+  inet_pton(AF_INET, "192.168.128.200", response->upf_endpoint.end_ipv4_addr);
+
+  response->always_on_pdu_session_indication     = false;
+  response->allowed_ssc_mode                     = SSC_MODE_3;
+  response->m5gsm_congetion_re_attempt_indicator = true;
+  response->pdu_address.redirect_address_type    = IPV4_1;
+  inet_pton(
+      AF_INET, "192.168.128.200",
+      response->pdu_address.redirect_server_address);
+}
+
+int send_pdu_session_response_itti() {
+  int rc                                          = RETURNerror;
+  itti_n11_create_pdu_session_response_t response = {};
+  create_pdu_session_response_ipv4_itti(&response);
+
+  imsi64_t imsi64;
+  IMSI_STRING_TO_IMSI64(response.imsi, &imsi64);
+  // Handle smf_context
+  ue_m5gmm_context_s* ue_context = lookup_ue_ctxt_by_imsi(imsi64);
+
+  if (!ue_context) {
+    return RETURNerror;
+  }
+
```

Comment:
Why is this part of send response function? It should be part of the test validation.

Reply:
removed above code, it was validating after initial UE Message handling.
```
  /* Check if UE Context is created with correct imsi */
  bool res = false;
  res      = get_ue_id_from_imsi(amf_app_desc_p, imsi64, &ue_id);
  EXPECT_TRUE(res == true);
```